phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -572,34 +577,40 @@ private void prepareToRebuildIndexes(Connection conn, String dataTableFullName)
             }
 
             if (hasViewIndex) {
-                ResultSet
-                        rs =
+                ResultSet rs =
                         conn.createStatement().executeQuery(
                                 "SELECT DISTINCT TABLE_NAME, TENANT_ID FROM "
-                                        + "SYSTEM.CATALOG WHERE COLUMN_FAMILY = \'"
-                                        + viewIndexPhysicalName
-                                        + "\' AND TABLE_TYPE = \'i\' AND " + "LINK_TYPE = "
+                                        + "SYSTEM.CATALOG "
+                                        + "WHERE COLUMN_FAMILY = \'"+dataTableFullName+"\' "
+                                        + "AND LINK_TYPE = "
                                         + PTable.LinkType.PHYSICAL_TABLE.getSerializedValue());
                 while (rs.next()) {
```

Comment:
I feel that the body of this while loop could be a helper func - findViewIndexes(string viewName). In fact, this should probably belong to some core Phoenix metadata classes for reuse. That we can do later, but for now, we should consider refactoring into another function.